src/Compilers/Core/Portable/NativePdbWriter/PdbWriter.cs

```diff
@@ -68,9 +72,9 @@ internal PdbLogger(bool logging)
             }
         }
 
-        private void MaybeFlush()
+        private void EnsureSpace(int space)
         {
-            if (_logData.Count >= bufferFlushLimit)
+            if (_logData.Count + space >= bufferFlushLimit)
```

Comment:
I'd add a comment that the underlying buffer can still be resized if the input is greater than bufferFlushLimit thus we use BlobBuilder rather than fixed sized BlobWriter.